MMSApplication/src/main/java/com/javaschool/onlineshop/service/impl/CartServiceImpl.java

```diff
@@ -97,9 +104,20 @@ public Double countTotal() {
         return cartElementDTOList;
     }
 
+
     @Transactional
-    public List<CartElement> getCartElements(){
+    public CartElement getCartElementById(Long id) {
+        return cartElementDAO.get(id);
+    }
+
+    @Transactional
+    public void updateCartElement(Long cartElementId, Integer quantity) {
+        CartElement cartElement = cartElementDAO.get(cartElementId);
         Cart cart = this.getCart();
```

Comment:
It's better to call getCart in Dao class (cartDAO.getCartById(1L)). When you will add logic to getCart method with mapstruct, then you wil receive dto object.